### PR TITLE
HOTFIX Node creates identity on each run

### PR DESCRIPTION
**Pull request by Waldz**

### CODE REVIEW

File: cmd/mysterium_server/command_run/identity_handler.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	if err != nil || !ih.manager.HasIdentity(identity.Address) {
+		return identity, errors.New("identity not found in cache")
+	}
+
+	return identity, nil
+}
+
+func (ih *identityHandler) launchNew() (id identity.Identity, err error) {
+	// if all fails, create a new one
+	id, err = ih.manager.CreateNewIdentity("")
```
Comment (shroomist) [inline cmd/mysterium_server/command_run/identity_handler.go]: Perhaps it's not the best place to hardcode default empty passphrase.
maybe a getter function, to take it from config, context or elsewhere?

File: cmd/mysterium_server/command_run/factory.go [hunk 2]
```diff
@@ -1,34 +1,74 @@
...
+		dialogWaiterFactory: func(identity identity.Identity) (communication.DialogWaiter, dto_discovery.Contact) {
 			address := nats_discovery.NewAddressForIdentity(identity)
 			return nats_dialog.NewDialogWaiter(address), address.GetContact()
 		},
-		SessionManager: &session.Manager{
-			Generator: &session.Generator{},
+		sessionManagerFactory: func(vpnServerIp string) session.ManagerInterface {
+			return openvpn_session.NewManager(openvpn.NewClientConfig(
+				vpnServerIp,
+				options.DirectoryConfig+"/ca.crt",
```
Comment (tadovas) [inline cmd/mysterium_server/command_run/factory.go]: Don't hardcode slashes - the code is not portable between OSes. https://golang.org/pkg/path/filepath/#Join is your friend here.

File: cmd/mysterium_server/command_run/identity_handler.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	}
+
+	if id, err = ih.launchLast(); err == nil {
+		return id, err
+	}
+
+	return ih.launchNew()
+}
+
+func (ih *identityHandler) launchExisting(address string) (id identity.Identity, err error) {
```
Comment (ignasbernotas) [inline cmd/mysterium_server/command_run/identity_handler.go]: Launch sounds strange. I'd prefer "use" instead:)

Comment (Waldz) [reply cmd/mysterium_server/command_run/identity_handler.go]: Fixed

File: cmd/mysterium_server/command_run/identity_handler.go (hunk 1 above)
Comment (Waldz) [reply cmd/mysterium_server/command_run/identity_handler.go]: Moved to constant `nodeIdentityPassword`

File: cmd/mysterium_server/command_run/factory.go (hunk 2 above)
Comment (Waldz) [reply cmd/mysterium_server/command_run/factory.go]: Fixed in all codebase

**Comment by Waldz** [comment]
Fixes made, @tadovas, please check

**Pull request merged**